homeassistant/components/airzone/climate.py

```diff
@@ -0,0 +1,199 @@
+                zone_data,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class AirzoneClimate(AirzoneEntity, ClimateEntity):
+    """Define an Airzone sensor."""
+
+    def __init__(
+        self,
+        coordinator: AirzoneUpdateCoordinator,
+        entry: ConfigEntry,
+        system_zone_id: str,
+        zone_data: dict,
+    ) -> None:
+        """Initialize Airzone climate entity."""
+        super().__init__(coordinator, entry, system_zone_id, zone_data)
+        self._attr_name = f"{zone_data[AZD_NAME]}"
+        self._attr_unique_id = f"{entry.entry_id}_{system_zone_id}"
+        self._attr_supported_features = SUPPORT_TARGET_TEMPERATURE
+        self._attr_target_temperature_step = API_TEMPERATURE_STEP
+
+    async def async_update_hvac_params(self, params) -> None:
+        """Send HVAC parameters to API."""
+        try:
+            await self.coordinator.airzone.put_hvac(params)
+        except (AirzoneError, ClientConnectorError) as error:
+            _LOGGER.error(error)
+        else:
+            self.coordinator.async_set_updated_data(self.coordinator.airzone.data())
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set hvac mode."""
+        params = {
+            API_SYSTEM_ID: self.system_id,
+            API_ZONE_ID: self.zone_id,
+        }
+        if hvac_mode == HVAC_MODE_OFF:
+            params[API_ON] = 0
+        else:
+            if self.get_zone_value(AZD_MODES):
+                mode = HVAC_MODE_HASS_TO_LIB.get(hvac_mode)
+                if mode != self.get_zone_value(AZD_MODE):
+                    params[API_MODE] = mode
+            params[API_ON] = 1
+        _LOGGER.debug("Set hvac_mode=%s params=%s", hvac_mode, params)
+        await self.async_update_hvac_params(params)
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        params = {
+            API_SYSTEM_ID: self.system_id,
+            API_ZONE_ID: self.zone_id,
+            API_SET_POINT: temp,
+        }
+        _LOGGER.debug("Set temp=%s params=%s", temp, params)
+        await self.async_update_hvac_params(params)
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self.get_zone_value(AZD_TEMP)
+
+    @property
+    def current_humidity(self) -> int | None:
+        """Return the current temperature."""
+        return self.get_zone_value(AZD_HUMIDITY)
+
+    @property
+    def hvac_action(self) -> str | None:
+        """Return the current running hvac operation if supported."""
+        if self.get_zone_value(AZD_ON):
+            if self.get_zone_value(AZD_AIR_DEMAND) or self.get_zone_value(
+                AZD_FLOOR_DEMAND
+            ):
+                action = cast(
+                    str, HVAC_ACTION_LIB_TO_HASS.get(self.get_zone_value(AZD_MODE))
+                )
```

Comment:
```suggestion
                action = cast(
                    str, HVAC_ACTION_LIB_TO_HASS[self.get_zone_value(AZD_MODE)]
                )
```

Can the value be missing?

Reply:
You're right, it shouldn't be missing :P

Reply:
@bdraco removed get and cast in 5dabd97b36141ea79c08583466200ce19eebcb32

Reply:
There are a few `HVAC_MODE_LIB_TO_HASS.get`s as well. Can those be adjusted as well?

Reply:
> There are a few `HVAC_MODE_LIB_TO_HASS.get`s as well. Can those be adjusted as well?

Removed in a49e4385c6